vllm/v1/engine/processor.py

```diff
@@ -83,6 +85,18 @@ def _validate_lora(self, lora_request: Optional[LoRARequest]) -> None:
             raise ValueError(f"Got lora_request {lora_request} but LoRA is "
                              "not enabled!")
 
+    def _validate_guided_decoding(
+            self, params: Union[SamplingParams, PoolingParams]) -> None:
+        if not isinstance(params, SamplingParams):
+            return
+        if self.decoding_config.guided_decoding_backend != "xgrammar":
+            raise ValueError(
+                "Only xgrammar guided decoding is supported in V1.")
+        if (params.guided_decoding and params.guided_decoding.backend
+                and params.guided_decoding.backend != 'xgrammar'):
+            raise ValueError(
+                "Only xgrammar guided decoding is supported in V1.")
```

Comment:
Validation got added in two different spots at different times during development (and I think they were both me, oops!)

We should reconcile the changes here with the checks triggered in `core_client.py`

Reply:
This is now addressed with the latest commit